portal/src/graphql/portal/mutations/updateAppConfigMutation.ts

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { useMutation, gql } from "@apollo/client";
+
+import { client } from "../../portal/apollo";
+import { PortalAPIApp } from "../../../types";
+import { UpdateAppConfigMutation } from "./__generated__/UpdateAppConfigMutation";
+
+// relative to project root
+const UPDATE_FILE_PATH = "./authgear.yaml";
+
+const updateAppConfigMutation = gql`
+  mutation UpdateAppConfigMutation(
+    $appID: String!
+    $updateFile: AppConfigFile!
+  ) {
+    updateAppConfig(
+      input: { appID: $appID, updateFiles: [$updateFile], deleteFiles: [] }
+    ) {
+      id
+      rawAppConfig
+      effectiveAppConfig
+    }
+  }
+`;
+
+export function useUpdateAppConfigMutation(
+  appID: string
+): {
+  updateAppConfig: (appConfigYaml: string) => Promise<PortalAPIApp | null>;
```

Comment:
I guess this hook should do more and encode the given config to YAML string. It is more easy to use.